src/firefox/index.js

```diff
@@ -17,6 +17,82 @@ import {default as defaultFirefoxConnector, REMOTE_PORT} from './remote';
 
 const log = createLogger(__filename);
 
+// Flow Types
+
+import type {DefaultFirefoxConnectorFn} from './remote';
+import type {ChildProcess} from 'child_process';
+import type {
+  PreferencesAppName,
+  PreferencesGetterFn,
+} from './preferences';
+
+import type {ExtensionManifest} from '../util/manifest';
+
+export type FirefoxRunnerParams = {
+  binary?: string,
+  profile?: string,
+  "new-instance"?: boolean,
+  "no-remote"?: boolean,
+  "foreground"?: boolean,
+  "listen"?: number,
+  "binary-args"?: Array<string> | string,
+  "env"?: {
+    [key: string]: string
+  },
+  "verbose"?: boolean,
+};
+
+export type FirefoxRunnerResults = {
+  process: ChildProcess,
+  binary: string,
+  args: Array<string>,
+};
+
+export type FirefoxRunnerFn =
+  (params: FirefoxRunnerParams) => FirefoxRunnerResults;
```

Comment:
> Actually, this is the signature of this function:
> 
> https://github.com/mozilla-jetpack/node-fx-runner/blob/master/lib/run.js#L21
> 
> defined in the 'fx-runner' dependency, which returns this object (which corresponds to the > type defined as FirefoxRunnerResults) hehe:

but the [first line you linked to](https://github.com/mozilla-jetpack/node-fx-runner/blob/master/lib/run.js#L21), shows that `runFirefox()` returns a promise, no?

The signature to me looks like:

````
export type FirefoxRunnerFn =
  (params: FirefoxRunnerParams) => Promise<FirefoxRunnerResults>;
````